cmake/custom/eigen.cmake

```diff
@@ -1,2 +1,2 @@
-find_package(Eigen3 3.1.0 REQUIRED)
+find_package(Eigen3 3.3 REQUIRED NO_MODULE)
 include_directories(SYSTEM ${EIGEN3_INCLUDE_DIR})
```

Comment:
This is not needed anymore.

Reply:
Some strange behavior:


It works with
```
include_directories(${EIGEN3_INCLUDE_DIR})
```
but not with 
```
target_include_directories(mrcpp-shared PUBLIC ${EIGEN3_INCLUDE_DIR})
target_include_directories(mrcpp-static PUBLIC ${EIGEN3_INCLUDE_DIR})
```
and not with
```
target_link_libraries(mrcpp-shared PUBLIC Eigen3::Eigen)
target_link_libraries(mrcpp-static PUBLIC Eigen3::Eigen)
```

Reply:
OK, so the dependency has moved down to `mrcpp-objlib`:
```
target_include_directories(mrcpp-objlib
  PRIVATE
    ${PROJECT_SOURCE_DIR}/include
    ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_INCLUDEDIR}
    ${CMAKE_CURRENT_LIST_DIR}
  PUBLIC
    ${EIGEN3_INCLUDE_DIR}
  )
```
The `OBJECT` library builds the source files into object files, without linking them. So you get the object files and can decide if you want to link them statically or dynamically or both (as we do now). Dependencies move down from the library targets to the `OBJECT` library. Hence your error above.

A couple of letdowns:
- `target_link_libraries` can't be used with `OBJECT` libraries (it probably can with a more recent version of CMake, but it doesn't matter for us)
- The target exported by Eigen cannot be used with `target_include_directories` hence the use of the variable `EIGEN3_INCLUDE_DIR` defined by the same package...

Reply:
Alas, to get tests compiling you'll also need to add `${EIGEN3_INCLUDE_DIR}` to the `INTERFACE` include directories for `mrcpp-shared` and `mrcpp-static`[^1] (again, pity that Eigen as imported target is flaky...)
In `tests/CMakeLists.txt` you can remove
```
target_include_directories(mrcpp-tests
  PRIVATE
    ${PROJECT_BINARY_DIR}/include/MRCPP
  )
```

[^1] I am looking for a way to make `src/CMakeLists.txt` more compact, because now it's a bit bloated and repetitive, I must admit.